app/models/book.rb

```diff
@@ -1,2 +1,7 @@
 class Book < ApplicationRecord
+  has_many :stocks, class_name: "Shop::Stock", inverse_of: :book
+  has_many :shops, through: :stocks
+
+  has_many :not_empty_stocks, -> { where('value > 0') }, class_name: "Shop::Stock", inverse_of: :book
```

Comment:
What is the reason for that? why not regular scope:
```
class Stock
  scope :empty, -> { joins(:stocks_shops).where('stocks_shops.value = 0') }
```
and then:
`book.stocks.empty`
eager loading should work in this case as well, if that is a reason.